lib/modules/dosomething/dosomething_global/dosomething_global.module

```diff
@@ -530,12 +526,15 @@ function dosomething_global_redirect_node_anon() {
  * @param int $redirect_status
  */
 function dosomething_global_redirect_node_language($node, $language_code, $redirect_status = 302) {
-
   $languages = language_list();
 
   // Can only redirect to an published, translated version.
-  if (!empty($node->translations->data[$language_code]) && (bool)$node->translations->data[$language_code]['status']) {
```

Comment:
It's easier to understand the exit point, and why a 404 is happening when it follow a path. 
```
if (has translation && it's published) {
 // go to that published translation
} 
else {
 // 404 the user
}
```
vs
```
if (has translation) {
 if (not published) {
 // 404 
}  
// go to translation
```

in the second case it's less clear why we are going to the translation